### ISSUE TITLE
karma: extras: usePolling

### ISSUE THREAD

**Issue by michaelsbradleyjr** [insin/nwb#297]
This issue is a:

- [ ] Bug report
- [ ] Feature request
- [ ] Question / support request
- [ X ] Other

When making use of nwb's `npm run test:watch`, on macOS Sierra (10.12.4), node v7.9.0, I was getting very high CPU usage even when the node process was idle, i.e. just waiting to be triggered by a change to a source or test file.

I spent several hours trying to figure out what was happening. This helped:

`sudo fs_usage -f pathname -w`

Thousands of files were getting polled in `node_modules`, relative to my project's top-level directory. I tried to figure out whether it was karma or webpack, or both, and whether I could tell either or both to exclude `node_modules` ....... long story short, nothing I changed in the `extra:` fields for karma and webpack, in `nwb.config.js`, was having any significant effect.

Later, while digging around in some GitHub issues, I saw where one can set `usePolling: false` in one's karma config (that options seems not well documented) 

So, in `nwb.config.js`:

```js
module.exports = {
  ...,
  karma: {
    extra: {
      ...,
      usePolling: false
    },
  ...
};
```

Finally, that worked. Now, the node process for `npm run test:watch` sits idle at ~0% cpu usage, which is what one would expect, i.e. when it's not busy compiling or running tests.

I'm filing this issue since it may be helpful to others, and perhaps warrants a note in nwb's configuration documentation.

**Comment by michaelsbradleyjr** [comment]
Okay, so it seems that on macOS, `usePolling` is getting set to `true` at runtime:

https://github.com/karma-runner/karma/commit/66f50d7e584d4cbde820e70746be3f3378440fa8

That commit was about 3 1/2 years ago, and it seems that non-polling file watching on macOS works fine now, i.e. since I set `usePolling` to `false` and the autoWatch facility is picking up changes and w/ low impact on CPU usage, as I noted previously. I'll leave the karma maintainers a note.

**Comment by michaelsbradleyjr** [comment]
Filed an issue yesterday with karma-runner/karma:

https://github.com/karma-runner/karma/issues/2669

**Comment by loklaan** [comment]
@michaelsbradleyjr Wow, nice discovery. I hope it becomes default in upstream karma 👍 .

Would you mind updating the [docs](https://github.com/insin/nwb/blob/master/docs/Configuration.md#karma-configuration) on the master branch? I'll be sure to merge it in asap.

**Comment by michaelsbradleyjr** [comment]
What do you think is the best approach? Should I include a note under the [`extra` heading](https://github.com/insin/nwb/blob/master/docs/Configuration.md#extra-object-1)?

**Comment by loklaan** [comment]
Under the [karma intro paragraph](https://github.com/insin/nwb/blob/master/docs/Configuration.md#karma-configuration) with a "**Note:**" prefix, would be great. Thanks for getting confirmation

**Comment by michaelsbradleyjr** [comment]
See #298.

**Comment by loklaan** [comment]
Resolved by #298
